### PR TITLE
UI fixes to review details screen.

### LINKED ISSUE #1303: UI Fixes to Review Details screen
As requested in the review of PR #1294 :

1. can we increase the margins on the single review screen:
- On product link section at the top: 12pt top and bottom
- On the review section: 16pt above the avatar and 16pt below the review actions

2. The colour of the product title, reviewer name, and date look darker than the secondary gray (#3C3C3C)

Original design for reference
ios-review

### PR DESCRIPTION
**Pull request by ctarda**
Fixes #1303 

Before|After
-|-
|

## Changes
- Updated height of `NoteDetailsHeaderPlainTableViewCell`
- Add implementation of `estimatedHeightForRowAt` to `ReviewDetailsViewController`
- Update top and both margins of `NoteDetailsCommentTableViewCell`

## Testing
- As with other PRs related to Reviews, testing is a bit cubmersome (courtesy of some of the storyboards):
- Checkout the branch, run the tests
- Undo the feature flag. Because the NotificationsViewController is instantiated in a Storyboard, it will be necessary to touch `FeatureFlag.swift` to:
```
        case .reviews:
            return BuildConfiguration.current == .localDeveloper
```
and it will also be necessary to change the class of the first view controller in the Notifications.storyboard from NotificationsViewController to ReviewsViewController

- Navigate to the reviews tab, check the new margins and font colors.

To avoid accidentally merging this, the build associated to the PR will not reflect the changes.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by pmusolino** [comment]
Everything looks fine to me, but I get 14 tests that fails on my local machine. Most seem to be related to this PR https://github.com/woocommerce/woocommerce-ios/pull/1404

Schermata 2019-10-30 alle 11 44 29

**Comment by ctarda** [comment]
@pmusolino 🤔 those tests should don't fail anymore... at least we thought we had a fix for those. I'll reopen #1398 thanks!

File: WooCommerce/Classes/ViewRelated/Notifications/Cells/NoteDetailsCommentTableViewCell.xib [hunk 1]
```diff
@@ -145,8 +143,8 @@
                 </subviews>
                 <color key="backgroundColor" white="1" alpha="1" colorSpace="custom" customColorSpace="genericGamma22GrayColorSpace"/>
                 <constraints>
-                    <constraint firstItem="QbZ-zr-lT5" firstAttribute="top" secondItem="p0w-Eh-NgH" secondAttribute="topMargin" id="Rl1-g8-YkH"/>
-                    <constraint firstAttribute="bottomMargin" secondItem="QbZ-zr-lT5" secondAttribute="bottom" id="Vfz-Zj-1v5"/>
+                    <constraint firstItem="QbZ-zr-lT5" firstAttribute="top" secondItem="p0w-Eh-NgH" secondAttribute="topMargin" constant="4" id="Rl1-g8-YkH"/>
+                    <constraint firstAttribute="bottomMargin" secondItem="QbZ-zr-lT5" secondAttribute="bottom" constant="6" id="Vfz-Zj-1v5"/>
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Notifications/Cells/NoteDetailsCommentTableViewCell.xib]: these `constant` diffs are the actual xib changes right? the other rect diffs are just automatically generated

File: WooCommerce/Classes/ViewRelated/Notifications/Cells/NoteDetailsHeaderPlainTableViewCell.xib [hunk 2]
```diff
@@ -1,25 +1,23 @@
...
+            <rect key="frame" x="0.0" y="0.0" width="375" height="48"/>
             <autoresizingMask key="autoresizingMask"/>
             <tableViewCellContentView key="contentView" opaque="NO" clipsSubviews="YES" multipleTouchEnabled="YES" contentMode="center" preservesSuperviewLayoutMargins="YES" insetsLayoutMarginsFromSafeArea="NO" tableViewCell="TIK-Mv-tcP" id="orX-X0-Z67">
-                <rect key="frame" x="0.0" y="0.0" width="375" height="43.5"/>
+                <rect key="frame" x="0.0" y="0.0" width="375" height="48"/>
                 <autoresizingMask key="autoresizingMask"/>
                 <subviews>
                     <label opaque="NO" multipleTouchEnabled="YES" contentMode="left" insetsLayoutMarginsFromSafeArea="NO" text="Title" textAlignment="natural" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" id="gqG-oH-lM4">
-                        <rect key="frame" x="16" y="0.0" width="343" height="43.5"/>
+                        <rect key="frame" x="16" y="0.0" width="343" height="48"/>
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Notifications/Cells/NoteDetailsHeaderPlainTableViewCell.xib]: 🤔any actual changes in this xib?

Comment (ctarda) [reply WooCommerce/Classes/ViewRelated/Notifications/Cells/NoteDetailsHeaderPlainTableViewCell.xib]: Yes, the cell height

File: WooCommerce/Classes/ViewRelated/Notifications/Cells/NoteDetailsCommentTableViewCell.xib (hunk 1 above)
Comment (ctarda) [reply WooCommerce/Classes/ViewRelated/Notifications/Cells/NoteDetailsCommentTableViewCell.xib]: Yep, I think so

**Pull request merged**